src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.Figure;
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public double getRandomlength() {
+        return random.nextDouble();
+    }
```

Comment:
this method can be private (and placed after public methods)

Reply:
Why private?

Reply:
encapsulation principle. object that will use your FigureSupplier, shouldn't know anything about how you calculate or create you figures, it should get and know only default figure or some figure without other details.